### PR TITLE
Update Thread::Queue signature and add test

### PR DESCRIPTION
**Pull request by mtsmfm**

### CODE REVIEW

File: stdlib/builtin/thread.rbs [hunk 1]
```diff
@@ -1038,10 +1038,8 @@ class Thread::Queue < Object
   # Also aliased as: [size](Queue.downloaded.ruby_doc#method-i-size)
   def length: () -> Integer
 
-  def marshal_dump: () -> untyped
```
Comment (mtsmfm) [inline stdlib/builtin/thread.rbs]: I can't find any documentation for `Thread::Queue#marshal_dump`

https://ruby-doc.org/core-2.7.0/Queue.html

File: stdlib/builtin/thread.rbs [hunk 2]
```diff
@@ -1017,19 +1017,19 @@ class Thread::Queue < Object
   #           end
   #         }
   #         q.close
-  def close: () -> untyped
+  def close: () -> self
 
   # Returns `true` if the queue is closed.
-  def closed?: () -> bool
+  def closed?: () -> (true | false)
```
Comment (soutaro) [inline stdlib/builtin/thread.rbs]: Please use `bool` instead of `true | false`.

File: stdlib/builtin/thread.rbs [hunk 3]
```diff
@@ -1053,21 +1051,21 @@ class Thread::Queue < Object
   # 
   # Also aliased as: [deq](Queue.downloaded.ruby_doc#method-i-deq),
   # [shift](Queue.downloaded.ruby_doc#method-i-shift)
-  def pop: (*untyped args) -> untyped
+  def pop: (?(true | false) non_block) -> untyped
```
Comment (soutaro) [inline stdlib/builtin/thread.rbs]: `bool` seems better type for this.

**Comment by mtsmfm** [comment]
Updated and force pushed

**Pull request merged**